File: backend/app/ecommerce/services/tax-service/src/main/java/com/congthanh/taxservice/model/entity/AbstractAuditEntity.java
package com.congthanh.taxservice.model.entity;

import com.congthanh.taxservice.listener.CustomAuditingEntityListener;
import jakarta.persistence.Column;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.MappedSuperclass;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.LastModifiedBy;

import java.time.Instant;

@MappedSuperclass
@Data
@EntityListeners(CustomAuditingEntityListener.class)
public abstract class AbstractAuditEntity {

    @CreationTimestamp
    @Column(name = "created_at", updatable = false, nullable = false)
    private Instant createdAt;

    @CreatedBy
    @Column(name = "created_by", updatable = false, nullable = false)
    private String createdBy;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @LastModifiedBy
    @Column(name = "updated_by", nullable = false)
    private String updatedBy;

}

File: backend/app/management/services/organization-service/src/department/entities/abstract-audit.entity.ts
import { AuditEntity } from 'src/common/interfaces/audit-entity.interface';
import { Column, CreateDateColumn, UpdateDateColumn } from 'typeorm';

export abstract class AbstractAuditEntity implements AuditEntity {
  @CreateDateColumn({ name: 'created_at', update: false, type: 'time without time zone' })
  createdAt: Date;

  @Column({ name: 'created_by', nullable: false, update: false })
  createdBy: string;

  @UpdateDateColumn({ name: 'updated_at', type: 'timestamp without time zone' })
  updatedAt: Date;

  @Column({ name: 'updated_by', nullable: false })
  updatedBy: string;
}
